### DISCUSSION TITLE
Azure Repo Git not supported?

### DISCUSSION THREAD

**Discussion by SA-Egress** [aquasecurity/trivy#3829] [Q&A] [answered]
Hi,

I'm working within the ADO framework and running Trivy from self-hosted agents through ADO pipelines.

I am struggling to get Trivy repo sub commands to work which are complaining about authentication issues. The documentation seems to only discuss _Scanning Private Repositories_ for Git repositories and not Azure repositories. Example, only the GITHUB tokens are discussed in regards to authentication.

Is it not supported or am I missing something.

An example error when trying to authenticate to a Azure Git Repo:

`2023-03-13T14:10:28.312Z	INFO	Misconfiguration scanning is enabled
2023-03-13T14:10:28.312Z	INFO	Need to update the built-in policies
2023-03-13T14:10:28.312Z	INFO	Downloading the built-in policies...
39.14 KiB / 39.14 KiB [-----------------------------------------------------------] 100.00% ? p/s 0s2023-03-13T14:10:29.888Z	INFO	Secret scanning is enabled
2023-03-13T14:10:29.888Z	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-03-13T14:10:29.888Z	INFO	Please see also https://aquasecurity.github.io/trivy/v0.38/docs/secret/scanning/#recommendation for faster secret detection
2023-03-13T14:10:29.921Z	FATAL	repository scan error: scan error: unable to initialize a scanner: unable to initialize a repository scanner: git clone error: authentication required
`

**Comment by itaysk** [comment, marked as answer]
Repo scanning currently only support private repos in GitHub/GitLab

**Reply by SA-Egress** [reply]
Thanks for the reponse itaysk. Two questions:

1. Any plans to support Azure Repos.

2. What would the main difference be between scanning the whole file system from a checked out git repo as opposed to the scanning the repo itself? I imagine the repo scan would look at the various branches but in terms of files/checks performed... are they producing the same output?

**Reply by itaysk** [reply]
1. no plans currently. feel free to open an issue and is there's interest we will evaluate.
2. yes, they are similar. you can clone your repo and the do a `fs` scan

**Reply by SA-Egress** [reply]
Thanks itaysk,

Recently noticed that when doing a filesystem scan using the extension and doing the same scan via CLI - we're getting two different results. Both scans are happening at the root folder.

```
    - task: trivy@1
      inputs:
        path: ./${{ parameters.repo_name }}
        exitCode: 0

    - task: CmdLine@2
      displayName: "Run trivy file system scans"
      inputs:
        script: |
            trivy fs --ignore-unfixed --scanners license --exit-code 0 ./${{ parameters.repo_name}} --license-full
            trivy fs --ignore-unfixed --scanners vuln,config,secret --exit-code 0 ./${{ parameters.repo_name}} --severity CRITICAL --format template --template "@$(System.DefaultWorkingDirectory)/Commerce/Deployment/ADO/JUnit_Templates/trivy_junit.tpl" -o $(System.DefaultWorkingDirectory)/${{ parameters.repo_name }}/junit-filesystem-report-critical-${{ parameters.repo_name }}.xml
            trivy fs --ignore-unfixed --scanners vuln,config,secret --exit-code 0 ./${{ parameters.repo_name}} --severity HIGH --format template --template "@$(System.DefaultWorkingDirectory)/Commerce/Deployment/ADO/JUnit_Templates/trivy_junit.tpl" -o $(System.DefaultWorkingDirectory)/${{ parameters.repo_name }}/junit-filesystem-report-high-${{ parameters.repo_name }}.xml
            trivy fs --ignore-unfixed --scanners vuln,config,secret --exit-code 0 ./${{ parameters.repo_name}} --severity UNKNOWN --format template --template "@$(System.DefaultWorkingDirectory)/Commerce/Deployment/ADO/JUnit_Templates/trivy_junit.tpl" -o $(System.DefaultWorkingDirectory)/${{ parameters.repo_name }}/junit-filesystem-report-unknown-${{ parameters.repo_name }}.xml
            trivy fs --ignore-unfixed --exit-code 0 --format spdx-json --output $(Build.SourcesDirectory)/filesystem_SBOM-${{ parameters.repo_name }}.json ./${{ parameters.repo_name}}
```

The output from the extension is listing these three High Vulnerabilities.
![Trivy](https://github.com/aquasecurity/trivy/assets/125481053/1054679d-35cd-4ffd-8e69-8c8c863b77f8)

Yet our JUnit output is not showing them.

![Trivy2](https://github.com/aquasecurity/trivy/assets/125481053/808fe684-d2f5-477b-898c-3764e706b982)


Note - you may notice the --ignore-unfixed option we use when running the CLI commands. However, the Trivy extension shows those vuln as having a fix available so you wouldn't expect the difference to be due to that.

Thanks
